src/render/idyll-display/components/variable-view-v2.js

```diff
@@ -30,39 +33,47 @@ const VariableViewV2 = withContext(
 
       this.state = {
         rows: [],
-        error: null
+        error: null,
       };
 
       this.addVariable = this.addVariable.bind(this);
       this.getRows = this.getRows.bind(this);
       this.handleGridUpdate = this.handleGridUpdate.bind(this);
+      this._isMounted = false;
 
       props.context.onUpdate(() => {
         this.getRows();
       });
     }
 
     componentDidMount() {
+      this._isMounted = true;
       this.getRows();
     }
 
-    componentDidUpdate(prevProps) {
-      const { context } = this.props;
-
-      const prevVarNodes = prevProps.context.ast.children.filter(node =>
-        ALLOWED_TYPES.includes(node.type)
-      );
-      const currVarNodes = context.ast.children.filter(node =>
-        ALLOWED_TYPES.includes(node.type)
-      );
+    componentWillUnmount() {
+      this._isMounted = false;
+    }
 
-      if (
-        JSON.stringify(prevProps.context.context.data()) !==
-          JSON.stringify(context.context.data()) ||
-        JSON.stringify(prevVarNodes) !== JSON.stringify(currVarNodes)
-      ) {
-        // when a var/data node is added, context.data() has changed, or variable data is different
-        this.getRows();
+    componentDidUpdate(prevProps) {
+      if (this._isMounted) {
+        const { context } = this.props;
+
+        const prevVarNodes = prevProps.context.ast.children.filter((node) =>
+          ALLOWED_TYPES.includes(node.type)
+        );
+        const currVarNodes = context.ast.children.filter((node) =>
+          ALLOWED_TYPES.includes(node.type)
+        );
+
+        if (
+          JSON.stringify(prevProps.context.context.data()) !==
+            JSON.stringify(context.context.data()) ||
+          JSON.stringify(prevVarNodes) !== JSON.stringify(currVarNodes)
```

Comment:
It seems to be okay, but in the future we may want to think through how to avoid what could potentially be expensive comparisons by having some smarter tracking of context values/updates. 

No need to change now, just noting in case we do hit performance issues updating the variable rows.

Reply:
Ah gotcha, this is good to note!